backend-utils/src/lib.rs

```diff
@@ -144,8 +146,8 @@ fn extract_namespace(name: &str) -> String {
     namespace.relative_to(&Default::default()).to_string()
 }
 
-/// From an identity, get the namespaces of the symbols it references.
-fn referenced_namespaces<F: FieldElement>(
+/// From an algebraic expression visitable, get the namespaces of the symbols it references.
+fn referenced_namespaces_algebraic_expression<F: FieldElement>(
     expression_visitable: &impl ExpressionVisitable<AlgebraicExpression<F>>,
```

Comment:
Does this also work for
```suggestion
    expression_visitable: &impl Children<AlgebraicExpression<F>>,
```
?

I would like to retire ExpressionVisitable since it is a bit too complicated (I don't yet have a solution for mutably iterating, but at least for ref-iterating, I think Children is much more handy).

Reply:
Cool! I simplified the two functions by using `AllChildren`. The trait to extract the namespace seems a bit overkill to me.